core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.java

```diff
@@ -234,22 +236,20 @@ public void onBackPressed() {
 
   private void exitAddNoteDialog() {
     if (noteEdited) {
-      Fragment previousInstance = getActivity().getSupportFragmentManager()
-        .findFragmentByTag(ConfirmationAlertDialogFragment.TAG);
-
-      if (previousInstance == null) {
-        // Custom AlertDialog for taking user confirmation before closing note dialog in case of unsaved changes
-        DialogFragment newFragment = new ConfirmationAlertDialogFragment(sharedPreferenceUtil, TAG,
-          R.string.confirmation_alert_dialog_message);
-        newFragment.show(getActivity().getSupportFragmentManager(),
-          ConfirmationAlertDialogFragment.TAG);
-      }
+      alertDialogShower.show(KiwixDialog.NotesDiscardConfirmation.INSTANCE, () -> {
+        dismissAddNoteDialog();
+        return Unit.INSTANCE;
+      });
     } else {
       // Closing unedited note dialog straightaway
       dismissAddNoteDialog();
     }
   }
 
+  protected void injection(@NonNull CoreComponent coreComponent) {
```

Comment:
this doesn't need to be protected, just an artifact of being copy pasted. You can just inline it, the original method is just a convenience because Activities are the most common place you have to call `inject` manually

Reply:
Done. Please review.